pallets/pools/src/lib.rs

```diff
@@ -611,21 +647,92 @@ pub mod pallet {
 			);
 
 			ensure!(
-				min_epoch_time >= T::MinEpochTimeLowerBound::get()
-					&& challenge_time >= T::ChallengeTimeLowerBound::get()
-					&& max_nav_age <= T::MaxNAVAgeUpperBound::get(),
-				Error::<T>::PoolParameterBoundViolated
+				EpochExecution::<T>::try_get(pool_id).is_err(),
+				Error::<T>::InSubmissionPeriod
 			);
 
-			Pool::<T>::try_mutate(pool_id, |pool| -> DispatchResult {
-				let pool = pool.as_mut().ok_or(Error::<T>::NoSuchPool)?;
+			if changes.min_epoch_time == Change::NoChange
+				&& changes.max_nav_age == Change::NoChange
+				&& changes.tranches == Change::NoChange
+			{
+				// If there's an existing update, we remove it
+				// If not, this transaction is a no-op
+				if ScheduledUpdate::<T>::contains_key(pool_id) {
+					ScheduledUpdate::<T>::remove(pool_id);
+				}
+
+				return Ok(());
+			}
+
+			if let Change::NewValue(min_epoch_time) = changes.min_epoch_time {
+				ensure!(
+					min_epoch_time >= T::MinEpochTimeLowerBound::get()
+						&& min_epoch_time <= T::MinEpochTimeUpperBound::get(),
+					Error::<T>::PoolParameterBoundViolated
+				);
+			}
+
+			if let Change::NewValue(max_nav_age) = changes.max_nav_age {
+				ensure!(
+					max_nav_age <= T::MaxNAVAgeUpperBound::get(),
+					Error::<T>::PoolParameterBoundViolated
+				);
+			}
+
+			let pool = Pool::<T>::try_get(pool_id).map_err(|_| Error::<T>::NoSuchPool)?;
+
+			if let Change::NewValue(tranches) = &changes.tranches {
+				Self::is_valid_tranche_change(Some(&pool.tranches), &tranches)?;
+			}
+
+			let now = Self::now();
+
+			let update = ScheduledUpdateDetails {
+				changes: changes.clone(),
+				scheduled_time: now.saturating_add(T::MinUpdateDelay::get()),
+			};
+
+			if T::MinUpdateDelay::get() == 0 && T::UpdateGuard::released(&pool, &update, now) {
+				Self::do_update_pool(&pool_id, &changes)
```

Comment:
Here (and the other branch) is where we need to return the actual weight consumed